starter-code/src/functions-and-arrays.js

```diff
@@ -33,6 +60,16 @@ const wordsArr = [
   'palace'
 ];
 
+
+function averageWordLength(wordsArr) {
+  wordsArray.forEach(word => {
+    console.log(word.length)
+    sum+=word.length.reduce((a,b)=>a+b); 
+  });
+}
+console.log(sum/wordsArr.length); 
+
+
 // Unique Arrays
 const wordsUnique = [
```

Comment:
Vous êtes allées un peu trop loin. L'idée c'est d'utiliser indexOf (on vérifie que la string n'existe pas en utilisant indexOf(array[i])===-1 
function uniquifyArray(array) {
  var newArray = [];
  if (array.length === 0) {
    return undefined;
  }
  for (var i = 0; i < array.length; i++) {
    if (newArray.indexOf(array[i]) === -1) {
      newArray.push(array[i]);
    }
  }
  return newArray;
}